kinto/tests/test_views_collections.py

```diff
@@ -35,12 +35,15 @@ def test_collections_name_should_be_simple(self):
                           headers=self.headers,
                           status=400)
 
-    def test_unknown_bucket_raises_404(self):
+    def test_unknown_bucket_raises_403(self):
```

Comment:
According to the documentation: 

> Status code 403 responses are the result of the web server being configured to deny access, for some reason, to the requested resource by the client.

Which doesn't implies necessarily that the resource exists. It just implies that the user cannot have any information about the resource.

I like the idea to return 404 when the user has the right to know that the resource doesn't exists and 403 in case it don't have the right to know either if it exists or not.